crypto/sr25519_test.go

```diff
@@ -0,0 +1,79 @@
+package crypto
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestSr25519SignAndVerify(t *testing.T) {
+	kp, err := GenerateSr25519Keypair()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	msg := []byte("helloworld")
+	sig, err := kp.Sign(msg)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pub := kp.Public().(*Sr25519PublicKey)
+	ok := pub.Verify(msg, sig)
+	if !ok {
+		t.Fatal("Fail: did not verify sr25519 sig")
+	}
+}
+
+func TestSr25519PublicKeys(t *testing.T) {
+	kp, err := GenerateSr25519Keypair()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	priv := kp.Private().(*Sr25519PrivateKey)
+	kp2, err := NewSr25519Keypair(priv.key)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !reflect.DeepEqual(kp.Public(), kp2.Public()) {
+		t.Fatal("Fail: pubkeys do not match")
```

Comment:
Not necessary but might be helpful to include the pubkeys in the error if they do not match.